pytest_factoryboy/fixture.py

```diff
@@ -191,7 +191,7 @@ def inject_into_caller(name: str, function: Callable, locals_: dict[str, Any]) -
     # Therefore, we can just check for __qualname__ to figure out if we are in a class, and apply the @staticmethod.
     is_class_or_function = "__qualname__" in locals_
     if is_class_or_function:
-        function = staticmethod(function)
+        function = staticmethod(function)  # type: ignore[assignment]
```

Comment:
since the build is now failing because of a bug of mypy and this line, I would say to remove the `type: ignore[assignment]` from here, and change the mypy builds to do a "ignore_outcome = true" so that we won't block on a failure on mypy for now.